.travis.yml

```diff
@@ -5,109 +5,119 @@ sudo: false
 
-before_script:
-    - if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then COMMIT_RANGE=$TRAVIS_COMMIT_RANGE; else COMMIT_RANGE="HEAD~..HEAD"; fi;
-    - if [ $TASK_SCA == 1 ]; then export COMMIT_SCA_FILES=`git diff --name-only --diff-filter=ACMRTUXB $COMMIT_RANGE`; fi
-
-script:
-    - if [ $TASK_SCA == 1 ]; then ./check_trailing_spaces.sh || travis_terminate 1; fi
-    - if [ $TASK_SCA == 1 ] && [ -n "$COMMIT_SCA_FILES" ]; then vendor/bin/phpmd `echo $COMMIT_SCA_FILES | sed 's/ /,/g'` text phpmd.xml --exclude src/Resources,tests/Fixtures || travis_terminate 1; fi;
-    - if [ $TASK_SCA == 1 ]; then php php-cs-fixer fix --rules @PHP70Migration:risky,@PHP71Migration,native_function_invocation -q; fi
-    - if [ $TASK_SCA == 1 ]; then travis_retry composer update $DEFAULT_COMPOSER_FLAGS $COMPOSER_FLAGS --no-dev --prefer-stable; fi
-    - if [ $TASK_SCA == 1 ]; then $HOME/.composer/vendor/bin/composer-require-checker check composer.json --config-file=.composer-require-checker.json || travis_terminate 1; fi
-    - if [ $TASK_SCA == 1 ]; then travis_retry composer update $DEFAULT_COMPOSER_FLAGS $COMPOSER_FLAGS; fi
-
-    - if [ $TASK_TESTS == 1 ] && [ $TASK_TESTS_COVERAGE == 0 ]; then vendor/bin/phpunit --verbose; fi
-    - if [ $TASK_TESTS == 1 ] && [ $TASK_TESTS_COVERAGE == 1 ]; then phpdbg -qrr vendor/bin/phpunit --verbose --coverage-clover build/logs/clover.xml; fi
-
-    - if [ $TASK_SCA == 1 ]; then git checkout . -q; fi
-
-    - if [ $TASK_CS == 1 ]; then php php-cs-fixer --diff --dry-run -v fix; fi
-
-after_success:
-    - if [ $TASK_TESTS_COVERAGE == 1 ]; then php vendor/bin/coveralls -v; fi
-
-before_deploy:
-    # install box2
-    - curl -LSs http://box-project.github.io/box2/installer.php | php
-    - php box.phar --version
-
-    # ensure that deps will work on lowest supported PHP version
-    - composer config platform.php 2> /dev/null || composer config platform.php 5.6.0
-
-    # require suggested packages
-    - composer require --no-update symfony/polyfill-mbstring
-
-    # update deps to highest possible for lowest supported PHP version
-    - composer update $DEFAULT_COMPOSER_FLAGS --no-dev --prefer-stable
-
-    - composer info -D | sort
-
-    # build phar file
-    - php -d phar.readonly=false box.phar build
-
-deploy:
-    provider: releases
-    api_key:
-        secure: K9NKi7X1OPz898fxtVc1RfWrSI+4hTFFYOik932wTz1jC4dQJ64Khh1LV9frA1+JiDS3+R6TvmQtpzbkX3y4L75UrSnP1ADH5wfMYIVmydG3ZjTMo8SWQWHmRMh3ORAKTMMpjl4Q7EkRkLp6RncKe+FAFPP5mgv55mtIMaE4qUk=
-    file: php-cs-fixer.phar
-    skip_cleanup: true
-    on:
-        repo: FriendsOfPHP/PHP-CS-Fixer
-        tags: true
-        condition: $DEPLOY = yes
+jobs:
+  include:
+    -
+        stage: Static Code Analysis
+        php: 7.1
+        env: COMPOSER_FLAGS="--no-dev --prefer-stable"
+        install:
+            - travis_retry composer global require $DEFAULT_COMPOSER_FLAGS phpmd/phpmd:^2.4.3 mi-schi/phpmd-extension:^4.2 maglnet/composer-require-checker:^0.1.4
```

Comment:
yeah, as file in repo
having the file, one could easily reproduce Travis steps locally (which is what I do with every 2.x->2.y merge)